model/compile/QtiAssetCompiler/QtiItemAssetCompiler.php

```diff
@@ -49,29 +58,41 @@ public function extractAndCopyAssetFiles(Item $qtiItem, Directory $publicDirecto
         $xincludeLoader->load();
 
         $packedAssets = [];
-
         foreach ($assetParser->extract() as $type => $assets) {
             foreach ($assets as $key => $assetUrl) {
                 if ($this->isBlacklisted($assetUrl)) {
                     continue;
                 }
-
                 $packedAsset = $this->resolve($resolver, $assetUrl, $type);
-
                 $replacement = $this->getReplacementName($packedAsset);
                 $packedAsset->setReplacedBy($replacement);
 
                 if ($type != 'xinclude') {
-                    $this->copyAssetFileToPublicDirectory($publicDirectory, $packedAsset);
+                    $packedAsset = $this->replaceWithExternalSource($packedAsset, $qtiItem);
+                    if (!$this->getQtiItemAssetReplacer()->shouldBeReplaced($packedAsset)) {
+                        $this->copyAssetFileToPublicDirectory($publicDirectory, $packedAsset);
+                    }
```

Comment:
I think it's possible to check 1 time `shouldBeReplaced`, and depending on the result `copyAssetFileToPublicDirectory` or `replaceWithExternalSource`

Reply:
we should not run `copyAssetFileToPublicDirectory` if `shouldBeReplaced` is true

Reply:
I think he means something like:
if shouldBeReplaced => replace with external sources / else copy asset file to public dir 
instead of testing 2 times the shouldBeReplaced :)

Reply:
already fixed :)